NRLMMD-GEOIPS/data_fusion
# v1.6.0 data_fusion - VERSION RELEASE - layered and fusion functionality

NOTE: This started as v1.5.4 release, but v1.6.0 justified due to data_fusion support updates.

# Related Issues
required to close NRLMMD-GEOIPS/geoips#51

# Reviewer Instructions
* Please confirm most recent completed pull request is labeled "DEVELOPMENTAL UPDATE" or "VERSION RELEASE":
    * https://github.com/NRLMMD-GEOIPS/data_fusion/pulls?q=is%3Apr+is%3Aclosed
* Review "Files changed" tab
* Confirm appropriate testing was completed for these updates.

# Summary
Changes for version 1.6.0 update on repo data_fusion.
See NRLMMD-GEOIPS/geoips#51 for additional repositories included in this update.

# Testing Instructions

Install GEOIPS and clone v1.5.4-release branch for:
* data_fusion
* geoips
* Test datasets contained in $GEOIPS_TESTDATA_DIR
  * test_data_scat v1.6.0-release branch: https://github.com/NRLMMD-GEOIPS/test_data_scat/tree/v1.6.0-release
  * ABI/AHI/SEVIRI data (see below)

Manually obtain ABI, AHI, and SEVIRI test datasets, in the specific locations required in the tests/scripts/*.sh scripts.

This should put the ABI and AHI data in the correct place for the "tests/scripts/geo.sh" call:
```
$GEOIPS/tests/download_noaa_aws.sh goes16 2021 09 29 00 00 $GEOIPS_TESTDATA_DIR/test_data_fusion/data/goes16_20210929.0000/ $GEOIPS/setup/rclone_setup/rclone.conf
$GEOIPS/tests/download_noaa_aws.sh goes17 2021 09 29 00 00 $GEOIPS_TESTDATA_DIR/test_data_fusion/data/goes17_20210929.0000/ $GEOIPS/setup/rclone_setup/rclone.conf
$GEOIPS/tests/download_noaa_aws.sh himawari8 2021 09 29 00 00 $GEOIPS_TESTDATA_DIR/test_data_fusion/data/himawari8_20210929.0000/ $GEOIPS/setup/rclone_setup/rclone.conf
$GEOIPS_PACKAGES_DIR/geoips/tests/utils/uncompress_test_data.sh $GEOIPS_TESTDATA_DIR/test_data_fusion bunzip2
```

SEVIRI data can be obtained directly from EUMETSAT. 
```
Place in these directories:
* $GEOIPS_TESTDATA_DIR/test_data_fusion/data/msg1_20210929.0000
* $GEOIPS_TESTDATA_DIR/test_data_fusion/data/msg4_20210929.0000

Clone and build open source XRITDecompress Software (https://gitlab.eumetsat.int/open-source/PublicDecompWT.git) in order to read SEVIRI data.

GeoIPS Utility will clone and build XRITDecompress, and link to $GEOIPS_DEPENDENCIES_DIR/bin:
$GEOIPS/setup.sh setup_seviri 
```

Tests should return 0 with all test datasets available:
$GEOIPS_PACKAGES_DIR/data_fusion/tests/test_all.sh

# Output
```
$GEOIPS_PACKAGES_DIR/data_fusion/tests/test_all.sh

Mon Dec 19 16:57:46 PST 2022 Running pre, log: $HOME/geoproc/test_data/logs/20221220/20221220.005746_data_fusion/test_all_data_fusion.log

Mon Dec 19 16:57:46 PST 2022  Running  geo.sh
$HOME/geoproc/geoips_packages/data_fusion/tests/scripts/geo.sh Infrared-Gray
$HOME/geoproc/test_data/logs/20221220/20221220.005746_data_fusion/test_all_data_fusion.log_geo.sh__Infrared-Gray.log
        Return: 0

Mon Dec 19 16:58:34 PST 2022  Running  layered.sh
$HOME/geoproc/geoips_packages/data_fusion/tests/scripts/layered.sh
$HOME/geoproc/test_data/logs/20221220/20221220.005746_data_fusion/test_all_data_fusion.log_layered.sh.log
        Return: 0


Mon Dec 19 17:00:25 PST 2022  Running post, final results in $HOME/geoproc/test_data/logs/20221220/20221220.005746_data_fusion/test_all_data_fusion.log

Package: data_fusion
Total run time: 159 seconds
Number data types run: 2
Number data types failed: 0
```

# Post Merge Steps
Once this PR has been merged, v1.6.0 will be tagged/released on the data_fusion main branch.
